src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -457,11 +506,20 @@ doCreateUserTable tn@(TableName ttxt) mn = do
     inDb lcTables t =
       callDb "doCreateUserTable" $ \db -> do
         r <- qry db (tableLookupStmt lcTables) [SText t] [RText]
-        return $ case r of
-          -- if lowercase matching, no need to check equality
-          -- (wasn't needed before either but leaving alone for replay)
-          [[SText rname]] -> lcTables || rname == t
-          _ -> False
+        case r of
+          [[SText rname]] ->
+            case mbh of
+                -- if lowercase matching, no need to check equality
+                -- (wasn't needed before either but leaving alone for replay)
+                Nothing -> return (lcTables || rname == t)
+                Just bh -> do
+                    qry db
+                        "SELECT tablename FROM VersionedTableCreation WHERE createBlockheight <= ? AND lower(tablename) = lower(?);"
```

Comment:
This could be unified with the query in `tableExists`, and the `<` vs `<=` distinction isn't needed here with `-1`.

Reply:
Also, this could use a comment as to why the table names need to be lower cased; which is briefly "because in the past they were case sensitive, and now they're not."